### PR TITLE
Updated error return to remove new line

### LINKED ISSUE #814: Fix error message
https://github.com/openfga/openfga/blob/61864b6fa8d8acaf79162b462e6785fb2fce276e/pkg/typesystem/typesystem.go#L727

This is adding a new line to the error message:

```json
{
"code":"invalid_authorization_model",
"message":"undefined relation\nadmin does not appear as a relation in any of the directly related user types [type:\"org\"]"
}
```

### PR DESCRIPTION
**Pull request by Galzzly**
PR resolves #814 by updating the returned error to remove the use of `errors.Join` which added the unwanted newline (`\n`).

## Description
Updated line 727 of `pkg/typesystem/typesystem.go` so that the the error prefix is included in the `fmt.Errorf` options, rather than using `errors.Join`

## References
Relates to Issue #814

### CODE REVIEW

**Comment by rhamzeh** [comment]
Thanks @Galzzly!

**Comment by Galzzly** [comment]
Working on the failing test, but hitting the identical string comparison failure issue for the two error strings. Will update once I am able to. Apologies for the delay.

File: pkg/typesystem/typesystem.go [hunk 1]
```diff
@@ -723,8 +723,9 @@ func (t *TypeSystem) isUsersetRewriteValid(objectType, relation string, rewrite
 					return nil
 				}
 			}
+			// ErrRes := ErrUserNotRelatedToType{computedUserset, userTypes}
 
-			return errors.Join(ErrRelationUndefined, fmt.Errorf("%s does not appear as a relation in any of the directly related user types %v", computedUserset, userTypes))
+			return fmt.Errorf("%w %s does not appear as a relation in any of the directly related user types %s", ErrRelationUndefined, computedUserset, userTypes)
```
Comment (jon-whit) [inline pkg/typesystem/typesystem.go:728]:
```suggestion
			return fmt.Errorf("%w: %s does not appear as a relation in any of the directly related user types %s", ErrRelationUndefined, computedUserset, userTypes)
```

**Comment by Galzzly** [comment]
All checks are now passing, I can resync the repository and push the updated repo up this afternoon if necessary.

**Comment by jon-whit** [comment]
@Galzzly mind rebasing on main? Then we can merge.

**Comment by Galzzly** [comment]
@jon-whit - That should be all sorted now.

**Review by miparnisari — APPROVED** [review]

**Pull request merged**